### DISCUSSION TITLE
[Lesson 4, Lecture 5] Homomorphic Encryption for Input Privacy

### DISCUSSION THREAD

**Discussion by majauhar** [OpenMined/courses#168] [Course: Our Privacy Opportunity] [answered]
Hi @iamtrask! In the lecture, you told us that we use public-key cryptography for homomorphic encryption. But, since the processing of data only happens after encryption by the input source, can't we use symmetric key cryptography instead?

**Comment by ZSoumia** [comment, marked as answer]
Hi @majauhar ! It can work with Private key , in fact in HE any symmetric key HE can be transformed to a public key HE ( [ Rothblum theorem](https://people.csail.mit.edu/vinodv/6892-Fall2013/rothblum.pdf)

**Reply by majauhar** [reply]
Thanks for the help @ZSoumia
